### DISCUSSION TITLE
iCub Software 1.8.0 Released

### DISCUSSION THREAD

**Discussion by drdanz** [robotology/community#236] [Releases]
iCub 1.8.0 was released today and is now available for download at: https://github.com/robotology/icub-main/releases/. This release is a feature release and introduces several new features and important changes since iCub 1.6.

Important Changes
-----------------

### Configuration Files

* changed file names pattern: now all file has following pattern:
  `body_part-ebX-jA_B-type_of_file.xml`

* in files `body_part-ebX-jA_B-mc.xml`:
   * added `TIMEOUT` group with velocity parameter.
   * added `CONTROL` group for new management of PID configuration. With this
     new feature, the user can write more than one pid configuration in the file
     and can select one by writing its name in `CONTROL` group.

* in files `body_part-ebX-jA_B-mc_service.xml`:
   * in `ENCODER1` and `ENCODER2` groups added `resolution` and `numofnoisebits`
     parameters.
   * removed group `CONTROLLER`. The coupling matrices have been moved in
     machanical file.

* in files `body_part-ebX-jA_B-mec.xml`:
   * added following parameters in `GENERAL` group
     - `fullscalePWM`: Value of fullscale pwm. This values is equal for all
       joints, becuse it depend on type of board, ems, mc4plus, etc.
     - `ampsToSensor`:     Scale value for the ampsToSensor conversion factor.
     - `Gearbox_M2J`:      This is the total reduction factor between joint and
       encoder.
     - `Gearbox_E2J`:      This is the reduction factor between the joint and
       the encoder at joint. (Usually it is 1)
     - `useMotorSpeedFbk`: Indicate if the motor controller uses feedback from
       motors to compute speed. Allowed values are [0, 1]
   * added group `2FOC` wich contains all parameter regarding 2FOC board
     configuration. These parameters have been moved from `GENERAL` group to
     `2FOC` group. Parameters about 2foc configuration are:
     - `HasHallSensor`: Indicates if hall sensors are connected to 2FOC board
     - `HasTempSensor`: Indicates if temperature ensor  are connected to 2FOC
       board
     - `HasRotorEncoder`: Indicates if rotor has encoder
     - `HasRotorEncoderIndex`: Indicates if the rotor encoder has index
     - `HasSpeedEncoder`: It is as an alternative to rotorEncoder. It is not
       used for FOC controller.
     - `RotorIndexOffset`: Rotor index offset between first electric sector and
       encoder index [degree]
     - `MotorPoles`: Number of poles of motor
   * added group `COUPLINGS`; it contains `matrixJ2M` (joint to motor),
     `matrixM2J` (motor to joint), `matrixE2J` (Encoder to joint) parameters.
   * added `JOINTSET_CFG` group. it contains specification of each joint set.
   * added `LIMITS` group. Its parameters are :
     - `hardwareJntPosMax`: Max hardware position of joint [expressed in
       degrees]. Due to mechanical constraints.
     - `hardwareJntPosMin`: Min hardware position of joint [expressed in
       degrees].  Due to mechanical constraints.
     - `rotorPosMin`: (moved from `body_part-ebX-jA_B-mc.xml` file)
     - `rotorPosMax`: (moved from `body_part-ebX-jA_B-mc.xml` file)

**Comment by traversaro** [comment]
Please note that, following #173, if you want to use the iCub software version 1.8.0 on a physical iCub mounting a pc104 board, you will need to update the OS running on the pc104 to the 6.0 image based on Debian Jessie. Please refer to http://wiki.icub.org/wiki/The_Linux_on_the_pc104 for more information .
